src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocPackageCheck.java

```diff
@@ -107,24 +107,41 @@ public void visitToken(DetailAST ast) {
      * @return true if there is javadoc, false otherwise
      */
     private static boolean hasJavadoc(DetailAST ast) {
-        final boolean hasBefore = isJavadocComment(ast.getPreviousSibling());
-        // need to go 3 levels down the tree for annotation case
-        final boolean hasWithAnnotation = Optional.of(ast.getFirstChild())
-            .map(DetailAST::getFirstChild)
+        final boolean hasBefore = ast.getPreviousSibling() != null
+            && isJavadoc(ast.getPreviousSibling());
+        return hasBefore || hasJavadocAboveAnnotation(ast);
+    }
+
+    /**
+     * Checks javadoc existence in annotations list.
+     *
+     * @param ast package def
+     * @return true if there is a javadoc, false otherwise
+     */
+    private static boolean hasJavadocAboveAnnotation(DetailAST ast) {
+        final Optional<DetailAST> firstAnnotationChild = Optional.of(ast.getFirstChild())
             .map(DetailAST::getFirstChild)
-            .map(MissingJavadocPackageCheck::isJavadocComment)
-            .orElse(false);
-        return hasBefore || hasWithAnnotation;
+            .map(DetailAST::getFirstChild);
+        boolean result = false;
+        if (firstAnnotationChild.isPresent()) {
+            for (DetailAST child = firstAnnotationChild.get(); child != null;
+                 child = child.getNextSibling()) {
+                if (isJavadoc(child)) {
+                    result = true;
+                    break;
+                }
+            }
+        }
+        return result;
     }
 
     /**
      * Checks that ast is a javadoc comment.
+     *
```

Comment:
I do not mind ... but please move to separate commit.
Looks like just another idea for new Check.  Is this update from your IDE ?

Reply:
I do not understand what you mean

Reply:
Please do other items.

This is just a side note that, non functional change is mixed with functional - not good.

Reply:
In some PRs I was asked to add such line to increase javadoc readability. I removed it

Reply:
yes, and it is completely ok to such clean up .... but there is nuance ... we do not mix non-fuctional refactoring with functional fixes in same commit, they should be in different commits (can be send in same PR).
split a single commit into two is easy:
```
git reset HEAD~1
git add -p 
# follow interactive mode by simply anwering y to non functional change
git commit -m "minor: code cleanup ...."
git add . 
git commit -m "Issue ......"
....
```

Reply:
I know how to split commits :) There is just no clear policy/check is empty line is required in javadoc

Reply:
yes, we have no Check on this, so in code we have whatever. I think separation empty line is good.